pwndbg/disasm/arm.py

```diff
@@ -64,9 +48,32 @@ def condition(self, instruction):
             ARM_CC_LT: N != V,
             ARM_CC_GT: not Z and (N == V),
             ARM_CC_LE: Z or (N != V),
-        }.get(instruction.cc, None)
+        }.get(instruction.cs_insn.cc, None)
+
+        if cc is None:
+            return InstructionCondition.UNDETERMINED
+
+        return InstructionCondition.TRUE if bool(cc) else InstructionCondition.FALSE
+
+    def memory_string(self, instruction: PwndbgInstruction, op: EnhancedOperand) -> str:
+        segment = ""
+        parts = []
+
+        if op.mem.base != 0:
+            parts.append(instruction.cs_insn.reg_name(op.mem.base))
+
+        if op.mem.disp != 0:
+            parts.append("%#x" % op.mem.disp)
+
+        if op.mem.index != 0:
+            index = pwndbg.gdblib.regs[instruction.cs_insn.reg_name(op.mem.index)]
+            scale = op.mem.scale
+            parts.append(f"{index}*{scale:#x}")
+
+        return f"[{(', '.join(parts))}]"
 
-        return cc
+    def immediate_string(self, instruction, operand):
+        return "#" + super().immediate_string(instruction, operand)
 
 
 assistant = DisassemblyAssistant("arm")
```

Comment:
I just realized there doesn't seem to be a `DisassemblyAssistant` for `aarch64`... I don't think the `arm` one will be used for it.  Maybe we do `assistant = DisassemblyAssistant("aarch64")` below this line, similar to how the x86 one has two assistants, one for 32 bit and one for 64 bit?

Reply:
I think the differences between arm and Aarch64 are too significant to attempt to put them in one class - we would end up with a bunch of "if is aarch64/arm" checks all over the place. Capstone itself differentiates the two into different classes altogether, while it keeps x86 as the same class whether is 32-bit or 64-bit, allowing for the x86 enhancer to be used for both. 

I'll make a seperate Aarch64 file/enhancer and add some annotation handlers to it.